node/client/src/lib.rs

```diff
@@ -193,13 +195,19 @@ impl Client {
 
         let last_block = self.beacon_chain.chain.best_block();
         let last_shard_block = self.shard_client.chain.best_block();
-        let authorities = self
+        let next_index = current_index + 1;
+        let authorities = match self
             .beacon_chain
             .authority
             .read()
             .expect(POISONED_LOCK_ERR)
-            .get_authorities(last_block.body.header.index + 1)
-            .expect("Authorities should be present for given block to produce it");
+            .get_authorities(next_index) {
+                Ok(authority) => authority,
+                Err(e) => {
+                    warn!("Cannot find authority for index {}: {}", next_index, e);
+                    return BlockProductionResult::AuthorityNotFound { next_index }
```

Comment:
This is actually should not be happening. Block production should not be failing due to pruning removing authorities needed for the block production.

Reply:
Fixed